### PR TITLE
Nge

### PR DESCRIPTION
**Pull request by shr-code**
## Related Issue
- Next Greater Element using stack (#734 )

## Proposed Changes
I have added the following things: -

- Problem Statement
- Explanation
- Code
- Time Complexity & Space Complexity
- Screenshot of Output

The Next greater Element for an element x is the first greater element on the right side of x in the array. Elements for which no greater element exist, consider the next greater element as -1. Examples: For an array, the rightmost element always has the next greater element as -1.

I have update README.md according to the guideline.

This next greater element problem is solved by using stack data structure.
The Time and Space complexity as well as the explanation is provided in the readme along with the output screenshot.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots

output

### CODE REVIEW

**Comment by Lakhankumawat** [comment]
Make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.


///------------------Description -----------------------///

For contribution and more guidelines i would recommend you to go through this 
tutorial video of the project [here](https://youtu.be/7UgC3HDETRA), thanks.



///------------------Description -----------------------///

Let me show you how to check the check list , there should be no space inside brackets like [x]
 `[x]`
- [x] Checklist
- [ ] Or you can just check them from outside without doing any of these
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [ ] 🌟 ed the repo

1. Starr the repo if you haven't !
2. Have you watched this [video](https://youtu.be/kIFwqp9B7pQ) ?


///----------------------PR README --------------------///


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by Lakhankumawat** [comment]
Remove Spacing in file names and modify your readme according to mentioned guidelines

**Comment by shr-code** [comment]
Sir, the problem statement readme or the pr readme?

**Comment by shr-code** [comment]
Can you please check now?

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
Follow the readme guidelines! Refer other prs!

**Comment by shr-code** [comment]
Sorry, I am not getting what to change?

**Comment by shr-code** [comment]
@siddhi-244 @Lakhankumawat

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
There are no table of contents in the readme. Follow the readme guidelines! Refer other prs !

**Comment by shr-code** [comment]
anyone? @Lakhankumawat @siddhi-244

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
check the spelling of algorithm and explain the approach !

**Comment by shr-code** [comment]
typos corrected! The approach (is just the visualization of algorithm) which is already explained. @siddhi-244 @Lakhankumawat

**Comment by siddhi-244** [comment]
> typos corrected! The approach (is just the visualization of algorithm) which is already explained. @siddhi-244 @Lakhankumawat

Cool !

File: S-Stack/NextGreaterElement/nge.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		cout << s.top() << " --> " << -1 << endl;
+		s.pop();
+	}
+}
+
+/* Driver code */
+int main()
+{
+	int arr[] = { 11, 13, 21, 3 };
+	int n = sizeof(arr) / sizeof(arr[0]);
```
Comment (siddhi-244) [inline S-Stack/NextGreaterElement/nge.cpp:52]: Can you take the array and it's size as input ?

**Review by siddhi-244 — CHANGES_REQUESTED** [review]

**Review by siddhi-244 — APPROVED** [review]
LGTM

File: S-Stack/NextGreaterElement/nge.cpp (hunk 1 above)
Comment (shr-code) [reply S-Stack/NextGreaterElement/nge.cpp]: Done! @siddhi-244

File: S-Stack/NextGreaterElement/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+Input: elements of array 11 13 21 3
+Output: 
+11-->13
+13-->21
+21-->-1
+3-->-1
+```
+<a name ="algo"></a>
+## Algorithim
```
Comment (Lakhankumawat) [inline S-Stack/NextGreaterElement/README.md:22]: Spelling Mistake (Typo)

File: S-Stack/NextGreaterElement/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+
+<a name ="space"></a>
+**Auxiliary Space:** O(n) for stack.
+
+<a name ="testcase"></a>
+### Test Case
+
+![ngeoutput](https://user-images.githubusercontent.com/84399701/164373069-017a36e2-551f-40e5-babe-020ee2634a7f.png)
+
```
Comment (Lakhankumawat) [inline S-Stack/NextGreaterElement/README.md:51]: Remove this section .

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

File: S-Stack/NextGreaterElement/README.md (hunk 3 above)
Comment (shr-code) [reply S-Stack/NextGreaterElement/README.md:51]: ok

**Comment by shr-code** [comment]
@Lakhankumawat done!!

**Comment by shr-code** [comment]
done @Lakhankumawat

**Review by Lakhankumawat — APPROVED** [review]
LGTM ✨🎉

**Comment by Lakhankumawat** [comment]
Good Contribution @shr-code  keep learning keep contributing 🎀

Mention your merged Pr numbers in your next prs and if i found them great i'll  increase your level of contribution ,

**Comment by Lakhankumawat** [comment]
@all-contributors please add @shr-code code ,

**Pull request merged**